bayesian/utility/v2/component_analyses.py

```diff
@@ -190,3 +225,17 @@ def add_unknown_pkg_info(stack_recommendation: List, unknown_pkgs: Set[Package])
         unknowns.pop('given_version', None)
         stack_recommendation.append(unknowns)
     return stack_recommendation
+
+
+def is_pseudo_version(ecosystem: str, version: str) -> bool:
+    """Check if given version is a pseudo version for golang ecosystem.
+
+    :param ecosystem: str
+    :param version: str - clean package version.
+    :return: True is its pseudo version for golang otherwise False.
+    """
+    if ecosystem == "golang":
+        # It is pseudo if contains 14 numeric timestamp and 12 digits commit hash.
+        return len(re.findall(r'(\d{14}|[0-9a-zA-Z]{12})', version)) == 2
```

Comment:
between the timestamp and the hash, we have a '-'. You can use that for the complete regex

Reply:
Regex is fixed now to validate 0.0.0-timestamp-commithash and its moved to UTILs.